src/nomos/agent/json_writer.c

```diff
@@ -10,23 +10,33 @@
 #include "nomos_utils.h"
 #include <json-c/json.h>
 
+static void writeHighlightInfoToJson(GArray* theMatches, json_object *resultsArray) {
+    int currentLicence;
+    for (currentLicence = 0; currentLicence < theMatches->len; ++currentLicence) {
+      LicenceAndMatchPositions* theLicence = getLicenceAndMatchPositions(theMatches, currentLicence);
+
+      int highl;
+      for (highl = 0; highl < theLicence->matchPositions->len; ++highl) {
+        MatchPositionAndType* ourMatchv = getMatchfromHighlightInfo(theLicence->matchPositions, highl);
+        json_object *result = json_object_new_object();
+        json_object_object_add(result, "license", json_object_new_string(theLicence->licenceName));
```

Comment:
If the `-S` flag for highlight is not passed, the output does not contain the name of licenses found in the file. This breaks backward compatibility. Maybe check if the `OPTS_HIGHLIGHT_STDOUT` is set and branch to use new code or existing code.

Reply:
I'll look into it.